controllers/temporalschedule_controller.go

```diff
@@ -0,0 +1,295 @@
+
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	cluster := &v1beta1.TemporalCluster{}
+	err = r.Get(ctx, namespace.Spec.ClusterRef.NamespacedName(schedule), cluster)
+	if err != nil {
+		if apierrors.IsNotFound(err) && !schedule.ObjectMeta.DeletionTimestamp.IsZero() {
+			logger.Info("Cluster not found deleting schedule", "cluster", namespace.Spec.ClusterRef.NamespacedName(schedule))
+			// Two ways to get here:
+			//  - TemporalCluster has not been created yet. In this case, if the TemporalSchedule is deleted, no point in waiting for the TemporalCluster to be healthy.
+			//  - TemporalCluster existed at some point, but now is deleted. In this case, the underlying schedule in the Temporal server is already gone.
+			controllerutil.RemoveFinalizer(schedule, deletionFinalizer)
+			return reconcile.Result{}, nil
+		}
+
+		return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Cluster lookup", err)
+	}
+
+	if !cluster.IsReady() {
+		logger.Info("Skipping schedule reconciliation until referenced cluster is ready")
+
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	clientOpts := func(opt *temporalclient.Options) {
+		opt.Namespace = schedule.Spec.NamespaceRef.Name
+	}
+	client, err := temporal.GetClusterClient(ctx, r.Client, cluster, clientOpts)
+	if err != nil {
+		err = fmt.Errorf("can't create cluster client: %w", err)
+		return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Creating cluster client", err)
+	}
+	defer client.Close()
+
+	// Check if the resource has been marked for deletion
+	if !schedule.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.Info("Deleting schedule")
+
+		err := r.ensureScheduleDeleted(ctx, schedule, &client)
+		if err != nil {
+			return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Deleting schedule", err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Ensure the schedule have a deletion marker if the AllowDeletion is set to true.
+	r.ensureFinalizer(schedule)
+
+	request, err := temporal.ScheduleToCreateScheduleRequest(cluster, schedule)
+	if err != nil {
+		return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Constructing create schedule request", err)
+	}
+
+	_, err = client.WorkflowService().CreateSchedule(ctx, request)
+	if err != nil {
+		var scheduleAlreadyExistsError *serviceerror.WorkflowExecutionAlreadyStarted
+		ok := errors.As(err, &scheduleAlreadyExistsError)
+		if !ok {
+			err = fmt.Errorf("can't create \"%s\" schedule: %w", schedule.GetName(), err)
+			return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Creating schedule", err)
+		}
+
+		request, err := temporal.ScheduleToUpdateScheduleRequest(schedule)
+		if err != nil {
+			return r.handleError(ctx, schedule, v1beta1.ReconcileErrorReason, "Constructing update schedule request", err)
+		}
+
+		_, err = client.WorkflowService().UpdateSchedule(ctx, request)
```

Comment:
Before updating the schedule, shouldn't we compare it on the controller-side to prevent useless updates ?

Reply:
I didn't do this as I was concerned it would add unnecessary complexity especially as the temporal server tends to apply defaults on its side. Do you see this impacting performance?

Reply:
Ok let's keep it as-is in the first release. If we have any performance concerns we would be able to fix this in a future PR.